destination.go

```diff
@@ -61,29 +61,37 @@ func (d *Destination) Open(_ context.Context) (err error) {
 	return nil
 }
 
-func (d *Destination) Write(ctx context.Context, recs []opencdc.Record) (int, error) {
+func (d *Destination) Write(ctx context.Context, recs []opencdc.Record) (written int, err error) {
+	tx, err := d.db.BeginTxx(ctx, nil)
+	if err != nil {
+		return 0, fmt.Errorf("failed to begin transaction: %w", err)
+	}
+
+	//nolint:errcheck // will always error if committed, no need to check
+	defer tx.Rollback()
+
 	for i, rec := range recs {
 		switch rec.Operation {
 		case opencdc.OperationSnapshot:
-			if err := d.upsertRecord(ctx, rec); err != nil {
+			if err := d.upsertRecord(ctx, tx, rec); err != nil {
 				return i, err
 			}
 		case opencdc.OperationCreate:
-			if err := d.upsertRecord(ctx, rec); err != nil {
+			if err := d.upsertRecord(ctx, tx, rec); err != nil {
 				return i, err
 			}
 		case opencdc.OperationUpdate:
-			if err := d.upsertRecord(ctx, rec); err != nil {
+			if err := d.upsertRecord(ctx, tx, rec); err != nil {
 				return i, err
 			}
 		case opencdc.OperationDelete:
-			if err := d.deleteRecord(ctx, rec); err != nil {
+			if err := d.deleteRecord(ctx, tx, rec); err != nil {
 				return i, err
 			}
 		}
 	}
 
-	return len(recs), nil
+	return len(recs), tx.Commit()
```

Comment:
If the transaction is not committed successfully, then that means that no records have been written, but we will still return `len(recs)` and indicate all records have been written?

Reply:
Huh, I'd say return 0 if error on all paths, as the transaction would be rolled back. I'll change that.